tracing/opencensus/endpoint_options.go

```diff
@@ -11,6 +15,11 @@ type EndpointOptions struct {
 	// Attributes holds the default attributes which will be set on span
 	// creation by our Endpoint middleware.
 	Attributes []trace.Attribute
+
+	// GetSpanName holds the function to use for generating the span name
+	// from the information found in the incoming Request.
+	// Defaults to the name that the middleware was initialized with.
+	GetSpanName func(ctx context.Context) string
```

Comment:
Should we add a functional option for it? Like `WithSpanNameGetter`?

Reply:
Since the other options have them I would say whatever we decide to add here should have a corresponding functional option.

That said I feel like we're adding something with limited value and could improve.

Some gaps to consider:
1. we should allow for a name regardless of being set explictly or not to be overridable
2. we should allow for a name regardless of being set explictly or not to be kept as is
3. we should allow for attributes to be set given the metadata as found in ctx.

Given the above gaps, how about we do this:
```go
GetSpanDetails func(ctx context.Context, name string) (newName string, attrs []trace.Attribute)
```

Reply:
Works for me, although that would be inconsistent with how the official OpenCensus plugins allow setting a name for a span.

It's also a bit error prone, because if you only want to return attributes and not a name you have to be aware that the returned name will always override any previously set one.

Reply:
I don't think that is a concern as long as you in the handling of the function test for empty (which could make it revert to previously set)

We're not trying to be inline with regular OpenCensus plugins here because the Endpoint middleware is distinctly different from those (they have no concept of a local span middleware)